### DISCUSSION TITLE
Feature Request: Allow secrets in approved external pull requests

### DISCUSSION THREAD

**Discussion by blowfishpro** [community/community#26242] [Actions]
<p><strong>The problem:</strong></p>
<p>Secrets and external pull requests are fundamentally in tension.  If you allow CI runs of external pull requests to use secrets, someone could submit a PR that either does something malicious with those secrets directly in the CI configuration or leaks them to be used later.  If you don’t allow CI runs of external pull requests to use secrets, you often can’t run tests or other verification against them, increasing the risk when you merge.</p>
<p><strong>The (possible) solution:</strong></p>
<p>A malicions PR could abuse secrets if it were merged regardless.  What prevents that is that a project collaborator can look at the code before merging and verify that it’s not doing anything bad.</p>
<p>So really there needs to be a way to say “a trusted human has looked at this code and verified that it’s legit, please run CI with secrets”</p>
<p>PR approvals are an existing mechanism for this.  An approval basically means “a trusted human (collaborator) has looked at this code and verified that it does what it claims to do and meets all required standards”.  So allowing CI to run with secrets against an external pull request once it has been approved seems to solve the problem without introducing any new security risk.</p>
<p><strong>My specific use case:</strong></p>
<p>I develop open source extensions for closed source (and paid) software.  Distributing the libraries to build against publicly would constitute a breach of the license, so I can’t do that.  But I want others to be able to propose changes via pull request.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/blowfishpro">@blowfishpro</a> ,</p>
<p>Thank you for being here! As <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#using-encrypted-secrets-in-a-workflow" rel="nofollow noopener">doc</a> mentioned: <span>With the exception of </span>GITHUB_TOKEN<span>, secrets are not passed to the runner when a workflow is triggered from a forked repository.</span></p>
<p>For your feature_request,  you can raise a ticket in the appropriate place <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a> where you can talk with github engineer team directly.  </p>
<p>Thanks.</p>

**Comment by blowfishpro** [comment]
<p>Yeah, posted this about 2 minutes before I found the feedback form.  Feel free to remove if you want.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/blowfishpro">@blowfishpro</a> ,</p>
<p>Thanks for your reply! It’s recommended to mark it as resolved(solution), canot remove ticket from my side. :slight_smile:</p>

**Comment by brillout** [comment]
<p>I couldn’t find any feature request for this on <a href="https://github.com/github-community/community/discussions" class="inline-onebox" rel="noopener nofollow ugc">Discussions · github-community/community · GitHub</a></p>
<p>Maybe I just missed it?</p>

**Comment by aaronlippold** [comment]
Any updates on this?

**Comment by bruno-garcia** [comment]
Anyone found the actual feature request for this?
This has been painful for years now
